File: Problems/No.933_Number_of_Recent_Calls/number_of_recent_calls.py
class RecentCounter(object):

    def __init__(self):
        self.recent_calls = []

    def ping(self, t):
        """
        :type t: int
        :rtype: int
        """

        self.recent_calls.append(t)
        while (self.recent_calls[0] < t-3000):
            self.recent_calls.pop(0)
        return len(self.recent_calls)
        


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: Problems/No.933_Number_of_Recent_Calls/number_of_recent_calls.cpp
class RecentCounter {

private:
    std::queue<int> recent_calls;

public:
    int ping(int t) {
        
        recent_calls.push(t);
        while (recent_calls.front() < t-3000) recent_calls.pop();
        return recent_calls.size();

    }
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */
